tests/certification/pubsub/azure/servicebus/servicebus_test.go

```diff
@@ -0,0 +1,1136 @@
+					return false, nil
+				}),
+			)
+		}
+	}
+
+	testTtlPublishMessages := func(metadata map[string]string, sidecarName string, topicName string, messageWatchers ...*watcher.Watcher) flow.Runnable {
+		return func(ctx flow.Context) error {
+			// prepare the messages
+			messages := make([]string, numMessages)
+			for i := range messages {
+				messages[i] = fmt.Sprintf("partitionKey: %s, message for topic: %s, index: %03d, uniqueId: %s", metadata[messageKey], topicName, i, uuid.New().String())
+			}
+
+			// add the messages as expectations to the watchers
+			for _, messageWatcher := range messageWatchers {
+				messageWatcher.ExpectStrings(messages...)
+			}
+
+			// get the sidecar (dapr) client
+			client := sidecar.GetClient(ctx, sidecarName)
+
+			// publish messages
+			ctx.Logf("Publishing messages. sidecarName: %s, topicName: %s", sidecarName, topicName)
+
+			var publishOptions dapr.PublishEventOption
+
+			if metadata != nil {
+				publishOptions = dapr.PublishEventWithMetadata(metadata)
+			}
+
+			for _, message := range messages {
+				ctx.Logf("Publishing: %q", message)
+				var err error
+
+				if publishOptions != nil {
+					err = client.PublishEvent(ctx, pubsubName, topicName, message, publishOptions)
+				} else {
+					err = client.PublishEvent(ctx, pubsubName, topicName, message)
+				}
+				require.NoError(ctx, err, "error publishing message")
+			}
+			// Wait for double the TTL after sending the last message.
+			time.Sleep(time.Second * 20)
+			return nil
+		}
+	}
+
+	assertMessages := func(timeout time.Duration, messageWatchers ...*watcher.Watcher) flow.Runnable {
+		return func(ctx flow.Context) error {
+			// assert for messages
+			for _, m := range messageWatchers {
+				m.Assert(ctx, 25*timeout)
+			}
+
+			return nil
+		}
+	}
+
+	flow.New(t, "servicebus certification").
+
+		// Test : defaultMessageTimeToLiveInSec optional attribute, in order processing with single publisher/subscriber
+		// Run subscriberApplication app1
+		Step(app.Run(appID1, fmt.Sprintf(":%d", appPort+portOffset*5),
+			subscriberApplication(appID1, topicActiveName, consumerGroup1))).
+
+		// Run the Dapr sidecar with the component entitymanagement
+		Step(sidecar.Run(sidecarName1,
+			embedded.WithComponentsPath("./components/default_ttl"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset*5),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset*5),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset*5),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset*5),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+		Step(fmt.Sprintf("publish messages to topicToBeCreated: %s", topicActiveName), testTtlPublishMessages(metadata, sidecarName1, topicActiveName, consumerGroup1)).
+		Step("verify if app6 has recevied messages published to newly created topic", assertMessages(10*time.Second, consumerGroup1)).
```

Comment:
Somehow you should publish the messages -- without subscribing to the messages - make note of the list of expected messages.
Then wait at least 2 * the default TTL window, then subscribe to the topic to receive the messages and use `consumerGroup1.FailIfNotExpected` or something like that.

It's important you don't subscribe to the message until the TTL has expired. If  you subscribe too early messages are delivered to your app / to the watcher.. and that defeats the purpose.